lib/matplotlib/tests/test_matplotlib.py

```diff
@@ -42,3 +42,20 @@ def parse(key):
             set(matplotlib.rcsetup.interactive_bk))
     assert (set(parse('- non-interactive backends:\n')) ==
             set(matplotlib.rcsetup.non_interactive_bk))
+
+
+def test_importable_with__OO():
+    """
+    When using -OO or export PYTHONOPTIMIZE=2, docstrings are discarded,
+    this simple test may prevent something like issue #17970.
+    """
+    program = (
+        "import numpy as np; "
+        "import numpy.ma as ma; "
+        "import matplotlib as mpl; "
+        "import matplotlib.pyplot as plt; "
+        "import matplotlib.cbook as cbook; "
+        "import matplotlib.patches as mpatches"
+    )
+    cmd = '{} -OO -c "{}"'.format(sys.executable, program)
+    assert subprocess.call(cmd, shell=True) == 0
```

Comment:
Use a list, not a string.
```suggestion
    cmd = [sys.executable, '-OO', '-c', program]
    assert subprocess.call(cmd) == 0
```